### PR TITLE
Stripe Payouts

### LINKED ISSUE #223: [LARGE] Stripe payment to hosts
Research how we can pay hosts via Stripe

### LINKED ISSUE #431: account link to stripe for hosts
start the connected accounts workflow in stripe for new hosts who arent connected; make a menu tab and add the payout stats (if possible) for the connected user after the stripe accounts are connected.

### PR DESCRIPTION
**Pull request by nishantbalaji**
Closes #223
Closes #431

### CODE REVIEW

**Comment by nishantbalaji** [comment]
Test this concurrently with the PR in the Backend directory

**Comment by nishantbalaji** [comment]
Changes

- connect button on profile
- dashboard link on profile

**Review by aidenszeto — COMMENTED** [review]
I feel like the Payouts tab is a bit empty (only has one button on it) so we should experiment with adding more information to the page, ie. instructions on how to set up the Stripe connection. We should also play around with including it in the Settings tab which is currently empty.

Let's discuss this some more at the next meeting make a decision from there.

**Review by aidenszeto — COMMENTED** [review]
@nishantbalaji LGTM. Before we merge, could you add a comment on how to test/verify functionality? Not sure how Stripe manages testing transactions but we should make sure everything works in the testing environment first.

File: src/components/myAccount/payout.component.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
+import React, { useState } from 'react'
+import { useSelector, useDispatch } from "react-redux";
```
Comment (aidenszeto) [inline src/components/myAccount/payout.component.js:2]: Doesn't look like we're using `useDispatch` here - should remove from import

**Review by aidenszeto — CHANGES_REQUESTED** [review]

**Comment by nishantbalaji** [comment]
> I feel like the Payouts tab is a bit empty (only has one button on it) so we should experiment with adding more information to the page, ie. instructions on how to set up the Stripe connection. We should also play around with including it in the Settings tab which is currently empty.
> 
> Let's discuss this some more at the next meeting make a decision from there.

Yeah I agree its a bit empty, but I didn't know where else to specifically put it. I also thought the stripe dashboard would be an integrated view but since its in a separate page we could just drop it in one of the other pages.

**Comment by nishantbalaji** [comment]
> @nishantbalaji LGTM. Before we merge, could you add a comment on how to test/verify functionality? Not sure how Stripe manages testing transactions but we should make sure everything works in the testing environment first.

To test with the testing environment, the keys have to be the testing keys. I think only the prod environment uses the live keys so anything we do with the testing keys will use the testing environment.

**Comment by aidenszeto** [comment]
I see. For the testing keys will you still need a valid account to connect with Stripe then?

Also how will Stripe know how much to pay out the hosts btw? Is that something we pass in?

**Review by aidenszeto — APPROVED** [review]

**Review by aidenszeto — APPROVED** [review]

**Pull request merged**